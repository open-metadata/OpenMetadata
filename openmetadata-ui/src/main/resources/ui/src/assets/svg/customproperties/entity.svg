<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 4.25C10.2225 4.25 10.44 4.18402 10.625 4.0604C10.81 3.93679 10.9542 3.76109 11.0394 3.55552C11.1245 3.34995 11.1468 3.12375 11.1034 2.90552C11.06 2.6873 10.9528 2.48684 10.7955 2.32951C10.6382 2.17217 10.4377 2.06503 10.2195 2.02162C10.0012 1.97821 9.77505 2.00049 9.56948 2.08564C9.36391 2.17078 9.18821 2.31498 9.0646 2.49998C8.94098 2.68499 8.875 2.9025 8.875 3.125C8.875 3.42337 8.99353 3.70952 9.2045 3.9205C9.41548 4.13147 9.70163 4.25 10 4.25ZM10 2.5C10.1236 2.5 10.2445 2.53666 10.3472 2.60533C10.45 2.67401 10.5301 2.77162 10.5774 2.88582C10.6247 3.00003 10.6371 3.12569 10.613 3.24693C10.5889 3.36817 10.5294 3.47953 10.4419 3.56694C10.3545 3.65435 10.2432 3.71388 10.1219 3.73799C10.0007 3.76211 9.87503 3.74973 9.76082 3.70243C9.64662 3.65512 9.54901 3.57501 9.48033 3.47223C9.41166 3.36945 9.375 3.24861 9.375 3.125C9.37519 2.9593 9.44109 2.80043 9.55826 2.68327C9.67543 2.5661 9.8343 2.50019 10 2.5Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M16.875 8.875C16.6525 8.875 16.435 8.94098 16.25 9.0646C16.065 9.18821 15.9208 9.36391 15.8356 9.56948C15.7505 9.77505 15.7282 10.0012 15.7716 10.2195C15.815 10.4377 15.9222 10.6382 16.0795 10.7955C16.2368 10.9528 16.4373 11.06 16.6555 11.1034C16.8738 11.1468 17.1 11.1245 17.3055 11.0394C17.5111 10.9542 17.6868 10.81 17.8104 10.625C17.934 10.44 18 10.2225 18 10C18 9.70163 17.8815 9.41548 17.6705 9.2045C17.4595 8.99353 17.1734 8.875 16.875 8.875ZM16.875 10.625C16.7514 10.625 16.6305 10.5883 16.5278 10.5197C16.425 10.451 16.3449 10.3534 16.2976 10.2392C16.2503 10.125 16.2379 9.99931 16.262 9.87807C16.2861 9.75683 16.3457 9.64547 16.4331 9.55806C16.5205 9.47065 16.6318 9.41112 16.7531 9.38701C16.8743 9.36289 17 9.37527 17.1142 9.42258C17.2284 9.46988 17.326 9.54999 17.3947 9.65277C17.4633 9.75555 17.5 9.87639 17.5 10C17.4998 10.1657 17.4339 10.3246 17.3167 10.4417C17.1996 10.5589 17.0407 10.6248 16.875 10.625Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M3.125 8.875C2.9025 8.875 2.68499 8.94098 2.49998 9.0646C2.31498 9.18821 2.17078 9.36391 2.08564 9.56948C2.00049 9.77505 1.97821 10.0012 2.02162 10.2195C2.06503 10.4377 2.17217 10.6382 2.32951 10.7955C2.48684 10.9528 2.6873 11.06 2.90552 11.1034C3.12375 11.1468 3.34995 11.1245 3.55552 11.0394C3.76109 10.9542 3.93679 10.81 4.0604 10.625C4.18402 10.44 4.25 10.2225 4.25 10C4.25 9.70163 4.13147 9.41548 3.9205 9.2045C3.70952 8.99353 3.42337 8.875 3.125 8.875ZM3.125 10.625C3.00139 10.625 2.88055 10.5883 2.77777 10.5197C2.67499 10.451 2.59488 10.3534 2.54758 10.2392C2.50027 10.125 2.48789 9.99931 2.51201 9.87807C2.53613 9.75683 2.59565 9.64547 2.68306 9.55806C2.77047 9.47065 2.88183 9.41112 3.00307 9.38701C3.12431 9.36289 3.24997 9.37527 3.36418 9.42258C3.47838 9.46988 3.57599 9.54999 3.64467 9.65277C3.71334 9.75555 3.75 9.87639 3.75 10C3.74981 10.1657 3.68391 10.3246 3.56674 10.4417C3.44957 10.5589 3.2907 10.6248 3.125 10.625Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M5.13867 6.26367C5.36118 6.26367 5.57868 6.19769 5.76369 6.07408C5.94869 5.95046 6.09289 5.77476 6.17804 5.56919C6.26318 5.36362 6.28546 5.13742 6.24206 4.9192C6.19865 4.70097 6.0915 4.50051 5.93417 4.34318C5.77683 4.18584 5.57638 4.0787 5.35815 4.03529C5.13992 3.99188 4.91372 4.01416 4.70815 4.09931C4.50259 4.18446 4.32689 4.32865 4.20327 4.51366C4.07965 4.69866 4.01367 4.91617 4.01367 5.13867C4.01367 5.43704 4.1322 5.72319 4.34318 5.93417C4.55416 6.14515 4.8403 6.26367 5.13867 6.26367ZM5.13867 4.51367C5.26229 4.51367 5.38312 4.55033 5.4859 4.619C5.58868 4.68768 5.66879 4.78529 5.7161 4.8995C5.7634 5.0137 5.77578 5.13937 5.75166 5.2606C5.72755 5.38184 5.66802 5.49321 5.58061 5.58061C5.49321 5.66802 5.38184 5.72755 5.2606 5.75166C5.13937 5.77578 5.0137 5.7634 4.89949 5.7161C4.78529 5.66879 4.68768 5.58868 4.619 5.4859C4.55033 5.38312 4.51367 5.26229 4.51367 5.13867C4.51386 4.97297 4.57977 4.81411 4.69694 4.69694C4.81411 4.57977 4.97297 4.51386 5.13867 4.51367Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M14.8613 6.26367C15.0838 6.26367 15.3013 6.19769 15.4863 6.07408C15.6713 5.95046 15.8155 5.77476 15.9007 5.56919C15.9858 5.36362 16.0081 5.13742 15.9647 4.9192C15.9213 4.70097 15.8142 4.50051 15.6568 4.34318C15.4995 4.18584 15.299 4.0787 15.0808 4.03529C14.8626 3.99188 14.6364 4.01416 14.4308 4.09931C14.2252 4.18446 14.0495 4.32865 13.9259 4.51366C13.8023 4.69866 13.7363 4.91617 13.7363 5.13867C13.7363 5.43704 13.8549 5.72319 14.0658 5.93417C14.2768 6.14515 14.563 6.26367 14.8613 6.26367ZM14.8613 4.51367C14.9849 4.51367 15.1058 4.55033 15.2086 4.619C15.3113 4.68768 15.3914 4.78529 15.4388 4.8995C15.4861 5.0137 15.4984 5.13937 15.4743 5.2606C15.4502 5.38184 15.3907 5.49321 15.3033 5.58061C15.2159 5.66802 15.1045 5.72755 14.9833 5.75166C14.862 5.77578 14.7364 5.7634 14.6222 5.7161C14.5079 5.66879 14.4103 5.58868 14.3417 5.4859C14.273 5.38312 14.2363 5.26229 14.2363 5.13867C14.2365 4.97297 14.3024 4.81411 14.4196 4.69694C14.5368 4.57977 14.6956 4.51386 14.8613 4.51367Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M10 15.75C9.7775 15.75 9.55999 15.816 9.37498 15.9396C9.18998 16.0632 9.04578 16.2389 8.96064 16.4445C8.87549 16.65 8.85321 16.8762 8.89662 17.0945C8.94003 17.3127 9.04717 17.5132 9.20451 17.6705C9.36184 17.8278 9.5623 17.935 9.78052 17.9784C9.99875 18.0218 10.225 17.9995 10.4305 17.9144C10.6361 17.8292 10.8118 17.685 10.9354 17.5C11.059 17.315 11.125 17.0975 11.125 16.875C11.125 16.5766 11.0065 16.2905 10.7955 16.0795C10.5845 15.8685 10.2984 15.75 10 15.75ZM10 17.5C9.87639 17.5 9.75555 17.4633 9.65277 17.3947C9.54999 17.326 9.46988 17.2284 9.42258 17.1142C9.37527 17 9.36289 16.8743 9.38701 16.7531C9.41113 16.6318 9.47065 16.5205 9.55806 16.4331C9.64547 16.3456 9.75683 16.2861 9.87807 16.262C9.99931 16.2379 10.125 16.2503 10.2392 16.2976C10.3534 16.3449 10.451 16.425 10.5197 16.5278C10.5883 16.6305 10.625 16.7514 10.625 16.875C10.6248 17.0407 10.5589 17.1996 10.4417 17.3167C10.3246 17.4339 10.1657 17.4998 10 17.5Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M5.13867 13.7363C4.91617 13.7363 4.69866 13.8023 4.51366 13.9259C4.32865 14.0495 4.18446 14.2252 4.09931 14.4308C4.01416 14.6364 3.99188 14.8626 4.03529 15.0808C4.0787 15.299 4.18584 15.4995 4.34318 15.6568C4.50051 15.8142 4.70097 15.9213 4.9192 15.9647C5.13742 16.0081 5.36362 15.9858 5.56919 15.9007C5.77476 15.8155 5.95046 15.6713 6.07408 15.4863C6.19769 15.3013 6.26367 15.0838 6.26367 14.8613C6.26367 14.563 6.14515 14.2768 5.93417 14.0658C5.72319 13.8549 5.43704 13.7363 5.13867 13.7363ZM5.13867 15.4863C5.01506 15.4863 4.89422 15.4497 4.79144 15.381C4.68866 15.3123 4.60855 15.2147 4.56125 15.1005C4.51394 14.9863 4.50157 14.8606 4.52568 14.7394C4.5498 14.6182 4.60932 14.5068 4.69673 14.4194C4.78414 14.332 4.8955 14.2725 5.01674 14.2483C5.13798 14.2242 5.26365 14.2366 5.37785 14.2839C5.49205 14.3312 5.58966 14.4113 5.65834 14.5141C5.72702 14.6169 5.76367 14.7377 5.76367 14.8613C5.76349 15.027 5.69758 15.1859 5.58041 15.3031C5.46324 15.4202 5.30438 15.4861 5.13867 15.4863Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M14.8613 13.7363C14.6388 13.7363 14.4213 13.8023 14.2363 13.9259C14.0513 14.0495 13.9071 14.2252 13.822 14.4308C13.7368 14.6364 13.7145 14.8626 13.7579 15.0808C13.8014 15.299 13.9085 15.4995 14.0658 15.6568C14.2232 15.8142 14.4236 15.9213 14.6419 15.9647C14.8601 16.0081 15.0863 15.9858 15.2918 15.9007C15.4974 15.8155 15.6731 15.6713 15.7967 15.4863C15.9203 15.3013 15.9863 15.0838 15.9863 14.8613C15.9863 14.563 15.8678 14.2768 15.6568 14.0658C15.4458 13.8549 15.1597 13.7363 14.8613 13.7363ZM14.8613 15.4863C14.7377 15.4863 14.6169 15.4497 14.5141 15.381C14.4113 15.3123 14.3312 15.2147 14.2839 15.1005C14.2366 14.9863 14.2242 14.8606 14.2483 14.7394C14.2725 14.6182 14.332 14.5068 14.4194 14.4194C14.5068 14.332 14.6182 14.2725 14.7394 14.2483C14.8606 14.2242 14.9863 14.2366 15.1005 14.2839C15.2147 14.3312 15.3123 14.4113 15.381 14.5141C15.4497 14.6169 15.4863 14.7377 15.4863 14.8613C15.4861 15.027 15.4202 15.1859 15.3031 15.3031C15.1859 15.4202 15.027 15.4861 14.8613 15.4863Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M6.56825 7.39937C6.58521 7.463 6.62606 7.51764 6.68228 7.55193C6.73851 7.58621 6.80578 7.5975 6.87012 7.58345C6.91375 7.57277 6.95367 7.55048 6.98566 7.51895C7.01764 7.48741 7.0405 7.44781 7.0518 7.40434L7.97675 8.32926C8.08317 8.20053 8.20153 8.08216 8.33026 7.97573L7.40532 7.05084C7.45911 7.0368 7.50667 7.00518 7.54042 6.96102C7.57418 6.91685 7.59221 6.86266 7.59163 6.80708C7.59106 6.75149 7.57192 6.69769 7.53726 6.65422C7.5026 6.61076 7.4544 6.58014 7.40033 6.56721L6.70044 6.39766C6.65879 6.38756 6.61525 6.38835 6.57399 6.39994C6.53273 6.41152 6.49514 6.43352 6.46483 6.46382C6.43453 6.49412 6.41252 6.53171 6.40093 6.57296C6.38933 6.61422 6.38854 6.65777 6.39862 6.69942L6.56825 7.39937Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M13.4319 12.5998C13.419 12.5457 13.3884 12.4976 13.3449 12.4629C13.3015 12.4282 13.2477 12.4091 13.1921 12.4085C13.1365 12.408 13.0823 12.426 13.0382 12.4597C12.994 12.4935 12.9624 12.5411 12.9484 12.5948L12.0234 11.6699C11.917 11.7986 11.7986 11.917 11.6699 12.0234L12.5947 12.9482C12.541 12.9623 12.4934 12.994 12.4597 13.0381C12.426 13.0823 12.408 13.1365 12.4086 13.1921C12.4092 13.2477 12.4283 13.3015 12.463 13.3449C12.4976 13.3884 12.5458 13.419 12.5999 13.432C12.5999 13.432 13.2992 13.6014 13.2997 13.6015C13.3414 13.6111 13.3848 13.61 13.426 13.5984C13.4671 13.5869 13.5047 13.5652 13.5353 13.5354C13.5656 13.5051 13.5877 13.4675 13.5993 13.4262C13.6109 13.385 13.6116 13.3414 13.6016 13.2998L13.4319 12.5998Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M13.9217 10.5054C13.9571 10.561 14.0127 10.6007 14.0768 10.6162C14.1408 10.6317 14.2084 10.6219 14.2653 10.5886L14.8802 10.2136C14.9167 10.1913 14.947 10.16 14.968 10.1226C14.9889 10.0853 15 10.0431 15 10.0003C15 9.95741 14.9889 9.91527 14.968 9.87791C14.947 9.84054 14.9167 9.8092 14.8802 9.78689L14.2653 9.41189C14.218 9.38278 14.1623 9.37035 14.107 9.37656C14.0518 9.38278 14.0002 9.40729 13.9605 9.44619C13.9208 9.48509 13.8952 9.53615 13.8879 9.59125C13.8806 9.64635 13.8919 9.70233 13.92 9.75027H12.6123C12.6201 9.83264 12.625 9.91589 12.625 10.0003C12.625 10.0846 12.6201 10.1679 12.6123 10.2503H13.92C13.8972 10.289 13.8853 10.3332 13.8857 10.3781C13.886 10.423 13.8984 10.467 13.9217 10.5054Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M9.99977 7.375C10.0842 7.375 10.1674 7.37983 10.2498 7.38766V6.08065C10.2978 6.10878 10.3538 6.12005 10.409 6.11266C10.4641 6.10527 10.5152 6.07966 10.5541 6.03989C10.593 6.00012 10.6175 5.94848 10.6237 5.89319C10.6299 5.83789 10.6174 5.78212 10.5882 5.73474L10.2132 5.11987C10.1909 5.08326 10.1596 5.05301 10.1222 5.03202C10.0848 5.01103 10.0426 5 9.99978 5C9.95691 5 9.91476 5.01103 9.87738 5.03202C9.84 5.05301 9.80865 5.08326 9.78634 5.11987L9.41134 5.73474C9.38226 5.7821 9.36985 5.83783 9.37608 5.89305C9.38232 5.94828 9.40683 5.99984 9.44573 6.03953C9.48464 6.07922 9.53569 6.10477 9.59078 6.11211C9.64587 6.11945 9.70183 6.10816 9.74977 6.08004V7.38766C9.83215 7.37983 9.91538 7.375 9.99977 7.375Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M12.0234 8.32925L12.9484 7.40433C12.9597 7.4478 12.9825 7.4874 13.0145 7.51894C13.0465 7.55047 13.0864 7.57276 13.1301 7.58344C13.1945 7.59842 13.2622 7.58752 13.3186 7.5531C13.3751 7.51867 13.4158 7.46347 13.4319 7.39936L13.6016 6.69941C13.6116 6.65776 13.6108 6.61422 13.5993 6.57296C13.5877 6.5317 13.5656 6.49412 13.5353 6.46382C13.505 6.43352 13.4675 6.41152 13.4262 6.39994C13.3849 6.38835 13.3414 6.38756 13.2997 6.39765L12.5998 6.56721C12.5458 6.58016 12.4976 6.61079 12.463 6.65425C12.4283 6.69771 12.4092 6.7515 12.4086 6.80708C12.408 6.86267 12.426 6.91685 12.4597 6.96103C12.4934 7.00521 12.541 7.03686 12.5947 7.05094L11.6699 7.97573C11.7986 8.08215 11.917 8.20052 12.0234 8.32925Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M10 7.875C9.57972 7.875 9.16887 7.99963 8.81941 8.23313C8.46996 8.46663 8.19759 8.7985 8.03676 9.1868C7.87592 9.57509 7.83384 10.0024 7.91583 10.4146C7.99783 10.8268 8.20021 11.2054 8.4974 11.5026C8.79459 11.7998 9.17322 12.0022 9.58543 12.0842C9.99764 12.1662 10.4249 12.1241 10.8132 11.9632C11.2015 11.8024 11.5334 11.53 11.7669 11.1806C12.0004 10.8311 12.125 10.4203 12.125 10C12.125 9.43641 11.9011 8.89591 11.5026 8.4974C11.1041 8.09888 10.5636 7.875 10 7.875ZM10 11.625C9.67861 11.625 9.36443 11.5297 9.0972 11.3511C8.82997 11.1726 8.62169 10.9188 8.4987 10.6219C8.3757 10.3249 8.34352 9.9982 8.40622 9.68298C8.46893 9.36776 8.62369 9.07821 8.85095 8.85095C9.07821 8.62369 9.36776 8.46892 9.68298 8.40622C9.9982 8.34352 10.3249 8.3757 10.6219 8.4987C10.9188 8.62169 11.1726 8.82997 11.3511 9.0972C11.5297 9.36443 11.625 9.67861 11.625 10C11.6245 10.4308 11.4532 10.8439 11.1485 11.1485C10.8439 11.4532 10.4308 11.6245 10 11.625Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M6.07998 9.75026C6.10811 9.70231 6.1194 9.64634 6.11206 9.59125C6.10471 9.53615 6.07916 9.48509 6.03945 9.44619C5.99975 9.40729 5.94818 9.38278 5.89294 9.37656C5.83771 9.37035 5.78198 9.38278 5.73462 9.41188L5.11981 9.78688C5.08322 9.80919 5.05298 9.84054 5.032 9.8779C5.01102 9.91527 5 9.95741 5 10.0003C5 10.0431 5.01102 10.0853 5.032 10.1226C5.05298 10.16 5.08322 10.1913 5.11981 10.2136L5.73462 10.5886C5.79148 10.6222 5.85916 10.6322 5.92331 10.6167C5.98747 10.6012 6.04304 10.5612 6.07824 10.5054C6.10154 10.467 6.11401 10.423 6.11431 10.3781C6.11462 10.3331 6.10276 10.289 6.07998 10.2503H7.38766C7.37983 10.1679 7.37499 10.0846 7.37499 10.0003C7.37499 9.91588 7.37983 9.83262 7.38766 9.75026H6.07998Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M7.97675 11.6699L7.0518 12.5948C7.03776 12.5411 7.00616 12.4935 6.962 12.4597C6.91784 12.426 6.86366 12.408 6.80808 12.4085C6.7525 12.4091 6.6987 12.4282 6.65524 12.4629C6.61179 12.4976 6.58116 12.5457 6.56824 12.5998L6.39862 13.2998C6.38854 13.3414 6.38933 13.385 6.40093 13.4262C6.41252 13.4675 6.43453 13.5051 6.46484 13.5354C6.49544 13.5652 6.53304 13.5869 6.57421 13.5985C6.61537 13.61 6.65877 13.6111 6.70044 13.6015L7.40033 13.432C7.4544 13.419 7.50258 13.3884 7.53724 13.3449C7.5719 13.3015 7.59104 13.2477 7.59162 13.1921C7.5922 13.1365 7.57418 13.0823 7.54044 13.0382C7.5067 12.994 7.45916 12.9624 7.40538 12.9483L8.33025 12.0234C8.20153 11.917 8.08317 11.7986 7.97675 11.6699Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
<path d="M9.99975 12.6259C9.91536 12.6259 9.83212 12.6211 9.74975 12.6133V13.9208C9.7018 13.8927 9.64583 13.8815 9.59075 13.8888C9.53567 13.8962 9.48462 13.9217 9.44573 13.9614C9.40684 14.0011 9.38232 14.0527 9.37609 14.1079C9.36985 14.1631 9.38225 14.2188 9.41131 14.2662L9.78631 14.8811C9.80862 14.9177 9.83997 14.9479 9.87735 14.9689C9.91472 14.9899 9.95687 15.0009 9.99974 15.0009C10.0426 15.0009 10.0848 14.9899 10.1221 14.9689C10.1595 14.9479 10.1909 14.9177 10.2132 14.8811L10.5882 14.2662C10.6172 14.2188 10.6296 14.1631 10.6234 14.1079C10.6172 14.0527 10.5927 14.0011 10.5538 13.9614C10.5149 13.9217 10.4638 13.8962 10.4087 13.8888C10.3537 13.8814 10.2977 13.8927 10.2497 13.9208V12.6133C10.1674 12.6211 10.0841 12.6259 9.99975 12.6259Z" fill="#292929" stroke="#292929" stroke-width="0.3"/>
</svg>
