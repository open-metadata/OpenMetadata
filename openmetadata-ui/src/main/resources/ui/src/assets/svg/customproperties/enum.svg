<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M6.41422 9.5H3.08578C2.48706 9.5 2 9.01294 2 8.41422V5.08578C2 4.48706 2.48706 4 3.08578 4H6.41422C7.01294 4 7.5 4.48706 7.5 5.08578V8.41422C7.5 9.01294 7.01294 9.5 6.41422 9.5ZM3.08578 5C3.03847 5 3 5.03847 3 5.08578V8.41422C3 8.46153 3.03847 8.5 3.08578 8.5H6.41422C6.46153 8.5 6.5 8.46153 6.5 8.41422V5.08578C6.5 5.03847 6.46153 5 6.41422 5H3.08578Z" fill="#292929"/>
<path d="M17.5 6.5H9.5C9.22387 6.5 9 6.27613 9 6C9 5.72387 9.22387 5.5 9.5 5.5H17.5C17.7761 5.5 18 5.72387 18 6C18 6.27613 17.7761 6.5 17.5 6.5Z" fill="#292929"/>
<path d="M14.75 8.375H9.5C9.22387 8.375 9 8.15113 9 7.875C9 7.59887 9.22387 7.375 9.5 7.375H14.75C15.0261 7.375 15.25 7.59887 15.25 7.875C15.25 8.15113 15.0261 8.375 14.75 8.375Z" fill="#292929"/>
<path d="M6.41422 16.75H3.08578C2.48706 16.75 2 16.2629 2 15.6642V12.3358C2 11.7371 2.48706 11.25 3.08578 11.25H6.41422C7.01294 11.25 7.5 11.7371 7.5 12.3358V15.6642C7.5 16.2629 7.01294 16.75 6.41422 16.75ZM3.08578 12.25C3.03847 12.25 3 12.2885 3 12.3358V15.6642C3 15.7115 3.03847 15.75 3.08578 15.75H6.41422C6.46153 15.75 6.5 15.7115 6.5 15.6642V12.3358C6.5 12.2885 6.46153 12.25 6.41422 12.25H3.08578Z" fill="#292929"/>
<path d="M17.5 13.625H9.5C9.22387 13.625 9 13.4011 9 13.125C9 12.8489 9.22387 12.625 9.5 12.625H17.5C17.7761 12.625 18 12.8489 18 13.125C18 13.4011 17.7761 13.625 17.5 13.625Z" fill="#292929"/>
<path d="M14.75 15.625H9.5C9.22387 15.625 9 15.4011 9 15.125C9 14.8489 9.22387 14.625 9.5 14.625H14.75C15.0261 14.625 15.25 14.8489 15.25 15.125C15.25 15.4011 15.0261 15.625 14.75 15.625Z" fill="#292929"/>
</svg>
