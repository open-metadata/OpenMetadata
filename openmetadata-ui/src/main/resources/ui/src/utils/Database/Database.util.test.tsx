/*
 *  Copyright 2023 Collate.
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *  http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
import { OwnerLabel } from '../../components/common/OwnerLabel/OwnerLabel.component';
import { OperationPermission } from '../../context/PermissionProvider/PermissionProvider.interface';
import { TabSpecificField } from '../../enums/entity.enum';
import { DatabaseSchema } from '../../generated/entity/data/databaseSchema';
import {
  DatabaseFields,
  ExtraDatabaseDropdownOptions,
  getQueryFilterForDatabase,
  schemaTableColumns,
} from './Database.util';

jest.mock(
  '../../components/Entity/EntityExportModalProvider/EntityExportModalProvider.component',
  () => ({
    useEntityExportModalProvider: jest.fn().mockReturnValue({
      showModal: jest.fn(),
    }),
  })
);
jest.mock(
  '../../components/common/ManageButtonContentItem/ManageButtonContentItem.component',
  () => ({
    ManageButtonItemLabel: jest
      .fn()
      .mockImplementation(() => <div>ManageButtonItemLabel</div>),
  })
);

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: mockNavigate,
}));

jest.mock('../../utils/TableColumn.util', () => ({
  ownerTableObject: jest.fn().mockReturnValue([
    {
      title: 'label.owner-plural',
      dataIndex: 'owners',
      key: 'owners',
      width: 180,
      filterIcon: () => <div>FilterIcon</div>,
      render: () => (
        <OwnerLabel
          isCompactView={false}
          maxVisibleOwners={4}
          owners={[{ id: '1', name: 'John Doe', type: 'user' }]}
          showLabel={false}
        />
      ),
    },
  ]),
}));

describe('Database Util', () => {
  describe('getQueryFilterForDatabase', () => {
    it('should return the correct query filter', () => {
      const serviceType = 'mysql';
      const databaseName = 'mydatabase';

      const expectedFilter = JSON.stringify({
        query: {
          bool: {
            must: [
              {
                bool: {
                  should: [
                    { term: { serviceType: [serviceType.toLowerCase()] } },
                  ],
                },
              },
              {
                bool: {
                  should: [
                    { term: { 'database.name.keyword': [databaseName] } },
                  ],
                },
              },
            ],
          },
        },
      });

      const actualFilter = getQueryFilterForDatabase(serviceType, databaseName);

      expect(actualFilter).toEqual(expectedFilter);
    });
  });

  describe('Database Util - DatabaseFields', () => {
    it('should have the correct fields', () => {
      const expectedFields = `${TabSpecificField.TAGS}, ${TabSpecificField.OWNERS}, ${TabSpecificField.DOMAINS},${TabSpecificField.DATA_PRODUCTS}`;

      expect(DatabaseFields).toEqual(expectedFields);
    });
  });

  describe.skip('Database Util - schemaTableColumns', () => {
    it('should render the correct columns', () => {
      const record = {
        name: 'schema1',
        fullyQualifiedName: 'database.schema1',
        description: 'Schema 1 description',
        owners: [{ id: '1', type: 'user', name: 'John Doe' }],
        usageSummary: {
          weeklyStats: { percentileRank: 80, count: 10 },
          dailyStats: { count: 10 },
          date: new Date(),
        },
        database: { name: 'database', id: 'database', type: 'database' },
        service: { name: 'service1', id: 'service1', type: 'service' },
      } as DatabaseSchema;

      const expectedColumns = [
        {
          title: 'label.schema-name',
          dataIndex: 'name',
          key: 'name',
          width: 250,
          render: expect.any(Function),
        },
        {
          title: 'label.description',
          dataIndex: 'description',
          key: 'description',
          render: expect.any(Function),
        },
        {
          title: 'label.owner-plural',
          dataIndex: 'owners',
          key: 'owners',
          width: 180,
          render: expect.any(Function),
          filterIcon: expect.any(Function),
        },
        {
          title: 'label.usage',
          dataIndex: 'usageSummary',
          key: 'usageSummary',
          width: 120,
          render: expect.any(Function),
        },
      ];

      const columns = schemaTableColumns;

      expect(columns).toEqual(expectedColumns);

      // Test render functions
      const nameColumn = columns[0];
      const descriptionColumn = columns[1];
      const ownerColumn = columns[2];
      const usageColumn = columns[3];

      // Test render function for name column
      const nameRender = nameColumn.render;
      const nameRenderResult = nameRender && nameRender(record.name, record, 0);

      expect(nameRenderResult).toMatchSnapshot();

      // Test render function for description column
      const descriptionRender = descriptionColumn.render;
      const descriptionRenderResult =
        descriptionRender && descriptionRender(record.description, record, 0);

      expect(descriptionRenderResult).toMatchSnapshot();

      // Test render function for owner column
      const ownerRender = ownerColumn.render;
      const ownerRenderResult =
        ownerRender && ownerRender(record.owners, record, 0);

      expect(ownerRenderResult).toMatchSnapshot();

      // Test render function for usage column
      const usageRender = usageColumn.render;
      const usageRenderResult =
        usageRender && usageRender(record.usageSummary, record, 0);

      expect(usageRenderResult).toMatchSnapshot();
    });
  });

  describe('Database Util - ExtraDatabaseDropdownOptions', () => {
    it('should render import button when user has editAll permission', () => {
      const permission = {
        ViewAll: false,
        EditAll: true,
      } as OperationPermission;

      const result = ExtraDatabaseDropdownOptions(
        'databaseFqn',
        permission,
        false,
        mockNavigate
      );

      expect(result).toHaveLength(1);
      expect(result[0].key).toBe('import-button');
    });

    it('should render export button when user has viewAll permission', () => {
      const permission = {
        ViewAll: true,
        EditAll: false,
      } as OperationPermission;

      const result = ExtraDatabaseDropdownOptions(
        'databaseFqn',
        permission,
        false,
        mockNavigate
      );

      expect(result).toHaveLength(1);
      expect(result[0].key).toBe('export-button');
    });

    it('should render both button when user has viewAll & editAll permission', () => {
      const permission = {
        ViewAll: true,
        EditAll: true,
      } as OperationPermission;

      const result = ExtraDatabaseDropdownOptions(
        'databaseFqn',
        permission,
        false,
        mockNavigate
      );

      expect(result).toHaveLength(2);
      expect(result[0].key).toBe('import-button');
      expect(result[1].key).toBe('export-button');
    });

    it('should not render any buttons when user has neither viewAll nor editAll permission', () => {
      const permission = {
        ViewAll: false,
        EditAll: false,
      } as OperationPermission;
      const result = ExtraDatabaseDropdownOptions(
        'databaseFqn',
        permission,
        false,
        mockNavigate
      );

      expect(result).toHaveLength(0);
      expect(result).toStrictEqual([]);
    });

    it('should not render any buttons when the entity is deleted', () => {
      const permission = {
        ViewAll: true,
        EditAll: true,
      } as OperationPermission;
      const result = ExtraDatabaseDropdownOptions(
        'databaseFqn',
        permission,
        true,
        mockNavigate
      );

      expect(result).toHaveLength(0);
      expect(result).toStrictEqual([]);
    });
  });
});
