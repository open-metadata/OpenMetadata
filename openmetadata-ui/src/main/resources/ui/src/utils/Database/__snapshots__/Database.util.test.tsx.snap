// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Database Util Database Util - schemaTableColumns should render the correct columns 1`] = `
<div
  className="d-inline-flex w-max-90"
>
  <Link
    className="break-word"
    data-testid="schema1"
    to="/databaseSchema/database.schema1"
  >
    schema1
  </Link>
</div>
`;

exports[`Database Util Database Util - schemaTableColumns should render the correct columns 2`] = `
<RichTextEditorPreviewerNew
  markdown="Schema 1 description"
/>
`;

exports[`Database Util Database Util - schemaTableColumns should render the correct columns 3`] = `
<OwnerLabel
  isCompactView={false}
  maxVisibleOwners={4}
  owners={
    Array [
      Object {
        "id": "1",
        "name": "John Doe",
        "type": "user",
      },
    ]
  }
  showLabel={false}
/>
`;

exports[`Database Util Database Util - schemaTableColumns should render the correct columns 4`] = `" 80th label.pctile-lowercase"`;
